### PR TITLE
GCI: Remove /var/lib/docker/network

### PR DESCRIPTION
**Pull request by yujuhong**
This avoids running into corrupt network checkpoint issues.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37593)

### CODE REVIEW

**Comment by yujuhong** [comment]
Verified the fix manually.

**Comment by yujuhong** [comment]
@dchen1107, I suggest cherry-picking this to 1.5. WDYT?

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -409,6 +409,23 @@ function assemble-docker-flags {
   fi
 
   echo "DOCKER_OPTS=\"${docker_opts} ${EXTRA_DOCKER_OPTS:-}\"" > /etc/default/docker
+
+  # If using a network plugin, extend the docker configuration to always remove
+  # the network checkpoint to avoid corrupt checkpoints.
+  # (https://github.com/docker/docker/issues/18283).
+  if [[ "${use_net_plugin}" == "true" ]]; then
```
Comment (adityakali) [inline cluster/gce/gci/configure-helper.sh]: I don't fully understand the network plugin check. Why do we want to do this only in this case?
Even the case below. You would have to restart docker daemon for any flag change to take effect. Not just the network plugin one.

Comment (yujuhong) [inline cluster/gce/gci/configure-helper.sh]: If user wants to use docker networking (without any plugins), we can't freely nuke the checkpoint every time docker restarts. We'll need to be able to detect the corrupt checkpoint.

The restart is done immediately below, so I didn't do it again here.

Comment (yujuhong) [inline cluster/gce/gci/configure-helper.sh]: Combined both `if` blocks to make things more clear.

**Comment by dchen1107** [comment]
LGTM.

Justification for including this to 1.5 milestone:

1) This is known issue in docker since 1.9 release, and the workaround is removing network checkpoint file. 
2) The debian-based containervm image has the workaround right before docker startup. There is a regression when we moved to gci.

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (adityakali) [inline cluster/gce/gci/configure-helper.sh]: From what I understand, I think the problem of "corrupt network checkpoint" can happen with docker networking too, and not just when the network plugins are used. But I leave that to your judgement.

**Comment by adityakali** [comment]
lgtm.

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (yujuhong) [inline cluster/gce/gci/configure-helper.sh]: The corrupt docker network checkpoint can happen regardless whether you use the plugin. However, if you do use docker networking (i.e., don't use a separate network plugin), we can't remove the checkpoint arbitrarily without making sure that the checkpoint is corrupt. On the other hand, when you *do* use a network plugin (i.e., not relying on docker networking), the network checkpoints are useless to you and it's safe to be removed.

**Comment by dims** [comment]
@k8s-bot gci gke e2e test this

**Comment by yujuhong** [comment]
@k8s-bot kops aws e2e test this

**Comment by yujuhong** [comment]
Submit queue doesn't pick up this PR at all. I am going to close and re-open it to hopefully trigger some action.

**Comment by dims** [comment]
@yujuhong  issue for tracking this sort of issue kubernetes/contrib#2102

**Comment by yujuhong** [comment]
> @yujuhong issue for tracking this sort of issue kubernetes/contrib#2102

Thanks!

**Comment by yujuhong** [comment]
@kubernetes/test-infra-admins all the tests have rerun and passed (if you clicked on them). How can I get the PR noticed by the submit queue?

**Comment by krzyzacy** [comment]
@rmmh is this when we mucking with SQ?

**Comment by ixdy** [comment]
@k8s-bot test this

**Comment by saad-ali** [comment]
> Justification for including this to 1.5 milestone:
> 
> This is known issue in docker since 1.9 release, and the workaround is removing network checkpoint file.
> The debian-based containervm image has the workaround right before docker startup. There is a regression when we moved to gci.

Ack.

**Pull request merged**